tests/python/gpu/test_operator_gpu.py

```diff
@@ -5,6 +5,7 @@
 from test_operator import *
 from test_optimizer import *
 from test_random import *
+from test_sparse_operator import test_sparse_dot
```

Comment:
@stefanhenneking 
Did you run `nosetests --verbose tests/python/gpu/test_operator_gpu.py` after you added test_sparse_dot to gpu test suite? It fails on my p2 machine with the most recent version of sparse branch 0d8d6c4a61520a79f2be3cbfc298ff584a1da178 
The last test(`test_dot_csr(lhs_shape, (lhs_shape[0], rnd.randint(1, 10)), 'row_sparse', True, 0.05)`) fails because it's not implemented yet.  
```
terminate called after throwing an instance of 'dmlc::Error'
  what():  [17:40:43] src/engine/./threaded_engine.h:329: [17:40:43] src/operator/tensor/./dot-inl.h:580: DotCsrRspDnsImpl has not implemented dot(csr.T, rsp) = dns yet

Stack trace returned 10 entries:
[bt] (0) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZN4dmlc15LogMessageFatalD1Ev+0x3f) [0x7f133a7cde69]
[bt] (1) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZN5mxnet2op16DotCsrRspDnsImplIN7mshadow3gpuEEEvPNS2_6StreamIT_EERKNS_7NDArrayESA_NS_9OpReqTypeEbPNS_5TBlobE+0$
6ff) [0x7f133bdf1bed]
[bt] (2) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZN5mxnet2op12DotForwardExIN7mshadow3gpuEEEvRKN4nnvm9NodeAttrsERKNS_9OpContextERKSt6vectorINS_7NDArrayESaISC_E$
RKSB_INS_9OpReqTypeESaISH_EESG_+0x562) [0x7f133bdec4ff]
[bt] (3) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZNSt17_Function_handlerIFvRKN4nnvm9NodeAttrsERKN5mxnet9OpContextERKSt6vectorINS4_7NDArrayESaIS9_EERKS8_INS4_9$
pReqTypeESaISE_EESD_EPSJ_E9_M_invokeERKSt9_Any_dataS3_S7_SD_SI_SD_+0x91) [0x7f133aba34a4]
[bt] (4) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZNKSt8functionIFvRKN4nnvm9NodeAttrsERKN5mxnet9OpContextERKSt6vectorINS4_7NDArrayESaIS9_EERKS8_INS4_9OpReqType$
SaISE_EESD_EEclES3_S7_SD_SI_SD_+0xa6) [0x7f133b41effe]
[bt] (5) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZN5mxnet4exec18FComputeExExecutor3RunENS_10RunContextEb+0x6a) [0x7f133b460ea2]
[bt] (6) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(+0x2436816) [0x7f133b479816]
[bt] (7) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(+0x2437e3a) [0x7f133b47ae3a]
[bt] (8) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZNKSt8functionIFvN5mxnet10RunContextENS0_6engine18CallbackOnCompleteEEEclES1_S3_+0x6e) [0x7f133b446d38]
[bt] (9) /home/ubuntu/haibin-mxnet/python/mxnet/../../lib/libmxnet.so(_ZN5mxnet6engine14ThreadedEngine15ExecuteOprBlockENS_10RunContextEPNS0_8OprBlockE+0x1e9) [0x7f133b4527d5]
```